### PR TITLE
Fix bots not drinking or eating because of dirty movement flags after facing something

### PR DESCRIPTION
**Pull request by Scarabol**
## 🍰 Pullrequest

I just debugged into the non-drinking/eating bots. The problem is that the spell is aborted, because the bot thinks it's moving all the time. You can see that by targeting your bot and calling ".debug moveflag" with gm permissions.
image
The flags are there, because MoveSplineInit::Launch() sets them, when it is called by [Unit::SetFacingTo](https://github.com/cmangos/mangos-tbc/blob/1789a95a27de389ce5ed771046db98c7d68592da/src/game/Entities/Unit.cpp#L4647) in the FollowMovementGenerator::_setOrientation call.

### Proof

- None

### Issues

- fixes #2830
- fixes #3309

### How2Test

- Log into game, add bot, make bot go out of mana or low on health
- Error: Bot spams chat, but does not drink or eat
- Apply fix provided here
- Solution: Bot drinks and eats as expected

### Todo / Checklist

- [x] None

### CODE REVIEW

**Comment by obi23lipnik** [comment]
Tested out on my server, and it works 💯

**Comment by f0n1x** [comment]
I ported this fix to classic, and it does indeed resolve the drinking/eating issue.

**Comment by obi23lipnik** [comment]
@killerwife want to call attention to this PR; have been running with this change on my server with heavy bot use without issues.. It's getting a little annoying to have to cherry-pick the two commits every rebase 😓 

TL;DR: 
- [ ] Approved
- [ ] Merged

**Comment by killerwife** [comment]
Sorry, christmas rushes at work have kept me away from wow. I will write a comment on the problematic line of code.

File: src/game/Entities/Unit.cpp [hunk 1]
```diff
@@ -4647,7 +4647,7 @@ void Unit::SetFacingTo(float ori)
     init.Launch();
     // orientation change is in-place
     UpdateSplinePosition();
-    movespline->_Finalize();
+    DisableSpline();
```
Comment (killerwife) [inline src/game/Entities/Unit.cpp:4650]: Disable calls _Interrupt, I need to call _Finalize. The way this works is that on blizz end, orientation SMSG_MONSTER_MOVE is immediate and is the reason why its this way. I understand that prolly the "MOVEFLAG_SPLINE_ENABLED | MOVEFLAG_FORWARD" is required, so a new function is in order called EndSpline. Please test if a new func with _Finalize and the moveflag change does the trick.

Comment (obi23lipnik) [reply src/game/Entities/Unit.cpp:4650]: @killerwife  so, like: https://github.com/cmangos/mangos-tbc/commit/cc5eacc000d3ab9bc0d9235ab4a5cb804f683c9a ?

I haven't yet been able to test this on my live server, but it's the gist, right?

Comment (obi23lipnik) [reply src/game/Entities/Unit.cpp:4650]: aye, does do the trick ☝️

Comment (killerwife) [reply src/game/Entities/Unit.cpp:4650]: Amazing.

Comment (killerwife) [reply src/game/Entities/Unit.cpp:4650]: Add the commit to the PR then.

Comment (obi23lipnik) [reply src/game/Entities/Unit.cpp:4650]: I have no edit permissions on this PR 😞 

Opened a new one: https://github.com/cmangos/mangos-tbc/pull/647

**Comment by killerwife** [comment]
Merged in the amended one

**Comment by joltcan** [comment]
The fix works great! Nice find!
